next/components/forms/steps/Summary/Summary.tsx

```diff
@@ -0,0 +1,105 @@
+import { ErrorSchema, RJSFValidationError, StrictRJSFSchema } from '@rjsf/utils'
+import { getAllPossibleJsonSchemaProperties, JsonSchema } from '@utils/forms'
+import { JSONSchema7Definition } from 'json-schema'
+
+import SummaryStep from './SummaryStep'
+import { TransformedFormData, TransformedFormStep } from './TransformedFormData'
+
+interface SummaryProps {
+  formData: Record<string, JsonSchema>
+  formErrors: RJSFValidationError[][]
+  extraErrors: ErrorSchema
+  schema?: StrictRJSFSchema
+  onGoToStep: (step: number) => void
+}
+
+const Summary = ({ schema, formData, formErrors, extraErrors, onGoToStep }: SummaryProps) => {
+  const isFieldError = (schemaPath: string, fieldName: string): boolean => {
+    const errorProperty = `${schemaPath}.${fieldName}`
+    return formErrors.some((errors) => {
+      return errors.some((error) => error.property === errorProperty)
+    })
+  }
+
+  const getFieldData = (
+    label: string,
+    schemaPath: string,
+    isError: boolean,
+    fieldFormData?: JSONSchema7Definition,
+  ): TransformedFormData => {
+    return {
+      label,
+      value:
+        (fieldFormData && !Array.isArray(fieldFormData)) ||
+        (Array.isArray(fieldFormData) && fieldFormData.length > 0)
+          ? JSON.stringify(fieldFormData, null, '\t').replaceAll('"', '')
+          : '-',
+      schemaPath,
+      isError,
+    }
+  }
+
+  const getAllSchemaData = (
```

Comment:
we can do this in a followup, but would like this function to be tested in isolation on a couple examples of prefilled form data, with and without errors

what that implies is for the function to live outside of the component, perhaps in one of the util files, which can be done as part of this PR